src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectFaultServiceExtensions.cs

```diff
@@ -0,0 +1,143 @@
+        /// </param>
+        /// <param name="severity">
+        ///     The severity of the failure.
+        /// </param>
+        public static void RegisterFaultHandler2(this IProjectFaultHandlerService faultHandlerService, Task task, UnconfiguredProject project, ProjectFaultSeverity severity = ProjectFaultSeverity.Recoverable)
+        {
+            Requires.NotNull(faultHandlerService, nameof(faultHandlerService));
+
+            faultHandlerService.RegisterFaultHandler(task, s_defaultReportSettings, severity, project);
+        }
+
+        /// <summary>
+        ///     Attaches error handling to a task so that if it throws an unhandled exception,
+        ///     the error will be reported to the user.
+        /// </summary>
+        /// <param name="faultHandlerService">
+        ///     The <see cref="IProjectFaultHostHandler"/> that should handle the fault.
+        /// </param>
+        /// <param name="task">
+        ///     The task to attach error handling to.
+        /// </param>
+        /// <param name="project">
+        ///     The project related to the failure, if applicable. Can be <see langword="null"/>.
+        /// </param>
+        /// <param name="severity">
+        ///     The severity of the failure.
+        /// </param>
+        public static void RegisterFaultHandler2<TResult>(this IProjectFaultHandlerService faultHandlerService, Task<TResult> task, UnconfiguredProject project, ProjectFaultSeverity severity = ProjectFaultSeverity.Recoverable)
+        {
+            Requires.NotNull(faultHandlerService, nameof(faultHandlerService));
+
+            faultHandlerService.RegisterFaultHandler(task, s_defaultReportSettings, severity, project);
+        }
+
+        /// <summary>
+        ///      Executes the specified delegate in a safe fire-and-forget manner, prevent the project from 
+        ///      closing until it has completed.
+        /// </summary>
+        /// <param name="threadingService">
+        ///     The <see cref="IProjectThreadingService"/> that handles the fork.
+        /// </param>
+        /// <param name="asyncAction">
+        ///      The async delegate to invoke. It is invoked asynchronously with respect to the caller.
+        /// </param>
+        /// <param name="unconfiguredProject">
+        ///     The unconfigured project which the delegate operates on, if applicable. Can be <see langword="null"/>.
+        /// </param>
+        /// <param name="faultSeverity">
+        ///     Suggests to the user how severe the fault is if the delegate throws.
+        /// </param>
+        /// <param name="options">
+        ///     Influences the environment in which the delegate is executed.
+        /// </param>
+        public static void Fork2(this IProjectThreadingService threadingService, Func<Task> asyncAction, UnconfiguredProject unconfiguredProject, ProjectFaultSeverity faultSeverity = ProjectFaultSeverity.Recoverable, ForkOptions options = ForkOptions.Default)
```

Comment:
Could we have a more descriptive name than `Fork2`?

Reply:
Maybe `ForkWithDefaultFaultHander` or at least `SafeFork`?

Reply:
Sure, what do you think? (TBH, Fork2 is an obvious replacement for Fork)

Reply:
`ForkWithDefaultReportSettings` would be most descriptive which I am partial to, but if you really like `Fork2` for its terseness I can live with it.

Reply:
`Fork_final`?

Reply:
What does the _final_ mean?

Reply:
I refuse to explain my joke other than to say `Fork_final2` has a nice ring to it.

Reply:
Yeah I can play around with that name @jmarolf.

Reply:
The numerical suffix thing on interfaces seemed weird to me when I joined MS but I understand it now as meaning the interface has been extended through versioning and derives from N-1. That's not the case here. `Fork` is already a rebranding of `Run` if I understand correctly. How about `RunWithErrorReporting` or `ForkWithErrorReporting`?

Reply:
I did consider Run, but I thought that was confusing with `JoinableTaskFactory.Run` (which is equivalent to `IProjectThreadingService.ExecuteSynchronously`). This is really; 'RunAndForget'

Reply:
I like `RunAndForget`. Neither `Fork` nor `ForkWithDefaultReportSettings` make it obvious to me that its forgetting, though I guess the void return helps.

Reply:
How about this;

Fork2 -> RunAndForget
RegisterFaultHandler2 -> Forget.

Reply:
Mind you Fork is more than JoinableTaskFactory.Run because it will switch threads and hide locks, which is why it was probably named as such.

Reply:
I've fixed the names.